liferay-lima/liferay-portal
# LPS-202141 Only enable the tag and show the panel if you have permissions to add

This is a follow up PR from https://github.com/liferay-lima/liferay-portal/pull/4938 with changes asked.

I moved the PermissionUtil class to permission.util package, following the pattern of DiscussionPermissionUtil instead of SharingPermissionUtil

And I moved the SharingPermissionUtil to the same package route, to put all the PermissionUtil classes on the taglibs module following the same pattern.



To test the story:

- Add a saved content taglib to the liferay Home Page
- Create a new Site
- Create new Page on the site and add the saved content taglib
- create a new user and add as a member of the new site (but not of liferay)
- impersonate or log as the new user
	- on Liferay home page the taglib is disabled
	- the "my saved content" panel does not appear on the User panel
	- Go to the New site
	- page the taglib is enabled and can save and delete the saved content
	- the "my saved content" panel appears on the User panel


Commits:

- LPS-202141 Create Saved Content Permission to check permissions from personal panel
- LPS-202141 baseline
- LPS-202141 check add_entry permission on the personal panel to show the My Saved Content panel only if you have permission for it
- LPS-202141 only enable the Saved Content Tag if you have permission for the site to add an entry
- LPS-202141 move to permission util package
- LPS-202141 move class to follow same pattern on taglibs
